src/lint/linter/UberCheckstyleLinter.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/**
+ * This linter invokes checkstyle for verifying code style standards.
+ */
+class UberCheckstyleLinter extends ArcanistLinter {
+
+  private $script = null;
+
+  public function getInfoName() {
+    return 'Java checkstyle linter';
+  }
+
+  public function getLinterName() {
+    return 'CHECKSTYLE';
+  }
+
+  public function getInfoURI() {
+    return 'http://checkstyle.sourceforge.net';
+  }
+
+  public function getInfoDescription() {
+    return 'Use checkstyle to perform static analysis on Java code';
+  }
+
+  public function getLinterConfigurationName() {
+    return 'checkstyle';
+  }
+
+  public function getLinterConfigurationOptions() {
+    $options = array(
+      'checkstyle.script' => array(
+        'type' => 'string',
+        'help' => pht('Checkstyle script to execute'),
+        )
+      );
+    return $options + parent::getLinterConfigurationOptions();
+  }
+
+  public function setLinterConfigurationValue($key, $value) {
+    switch ($key) {
+      case 'checkstyle.script':
+      $this->script = $value;
+      return;
+    }
+    return parent::setLinterConfigurationValue($key, $value);
+  }
+
+  public function lintPath($path) {
+    if (array_search($path, $this->getPaths()) === 0) {
+      $chunks = array_chunk($this->getPaths(), 500);
+      $futures = id(new FutureIterator(array()));
+      foreach ($chunks as $chunk) {
+        $future = new ExecFuture('%C %C', $this->script, implode(' ', $this->getPaths()));
+        $future->setCWD($this->getProjectRoot());
+        $futures->addFuture($future);
+      }
+
+      foreach ($futures as $future) {
+        list($stdout) = $future->resolvex();
+        $this->parseCheckstyleOutput($stdout);
+      }
+    }
+    return;
+  }
+
+  private function parseCheckstyleOutput($stdout) {
+    $dom = new DOMDocument();
+    $ok = @$dom->loadXML($stdout);
```

Comment:
I believe you still needed `@dom->loadXML($stdout);` part. Just `$ok` variable was unnecessary.